core/src/main/java/org/apache/iceberg/util/DateTimeUtil.java

```diff
@@ -86,6 +83,10 @@ public static long microsFromTimestamptz(OffsetDateTime dateTime) {
   }
 
   public static String formatTimestampMillis(long millis) {
-    return DATE_FORMAT.format(LocalDateTime.ofInstant(Instant.ofEpochMilli(millis), ZoneOffset.UTC));
+    return Instant.ofEpochMilli(millis)
+        .atOffset(ZoneOffset.UTC)
+        .toZonedDateTime()
+        .toString()
+        .replace("Z", "+00:00");
```

Comment:
Why not keep `Z`? It's a standard designation for UTC, I think :thinking:

Reply:
see https://github.com/apache/iceberg/issues/5321#issuecomment-1191968188 as a reference.

Reply:
ok, both notations are standard, but `Z` is simpler, IMHO :)